include/bluetooth/gatt_pool.h

```diff
@@ -66,30 +66,37 @@ extern "C" {
 /** @brief Register a characteristic descriptor.
  *
  *  @param _gp GATT service object with dynamic attribute allocation.
- *  @param _chrc_uuid_init Characteristic UUID.
- *  @param _char_props Properties of the characteristic.
- */
-#define BT_GATT_POOL_CHRC(_gp, _chrc_uuid_init, _char_props)                   \
+ *  @param _uuid Characteristic UUID.
+ *  @param _props Characteristic properties.
+ *  @param _perm Characteristic access permissions.
+ *  @param _read Characteristic read callback.
+ *  @param _write Characteristic write callback.
+ *  @param _value Characteristic value.
+ */
+#define BT_GATT_POOL_CHRC(_gp, _uuid, _props, _perm, _read, _write, _value)    \
 	do {                                                                   \
 		int _ret;                                                      \
-		const struct bt_gatt_chrc _chrc = {                            \
-		    .uuid = _chrc_uuid_init,                                   \
-		    .properties = _char_props,                                 \
-		};                                                             \
-		_ret = bt_gatt_pool_chrc_alloc(_gp, &_chrc);                   \
+		const struct bt_gatt_attr _attr = BT_GATT_ATTRIBUTE(           \
+			_uuid, _perm, _read, _write, _value);                  \
+		_ret = bt_gatt_pool_chrc_alloc(_gp, _props, &_attr);           \
 		__ASSERT_NO_MSG(!_ret);                                        \
 		(void)_ret;                                                    \
 	} while (0)
 
 /** @brief Register an attribute descriptor.
  *
  *  @param _gp GATT service object with dynamic attribute allocation.
- *  @param _descriptor_init Attribute descriptor.
+ *  @param _uuid Descriptor UUID.
+ *  @param _perm Descriptor access permissions.
+ *  @param _read Descriptor read callback.
+ *  @param _write Descriptor write callback.
+ *  @param _value Descriptor value.
  */
-#define BT_GATT_POOL_DESC(_gp, _descriptor_init)                               \
```

Comment:
Well... The old one style is a little more universal - you may initialize it inline:
```
BT_GATT_POOL_DESC(gp, BT_GATT_DESCRIPTOR(...));
```
That is a sweetener for direct initialisation: 
```
BT_GATT_POOL_DESC(gp, {.uuid = ..., .val = ...});
```

But it was also possible to use a previously initialized element:
```
bt_gatt_attr attr = BT_GATT_DESCRIPTOR(...);

BT_GATT_POOL_DESC(gp, attr);
```

So I am not sure that the idea presented here is right way to go.

Reply:
I did this change to align BT_GATT_POOL_DESC with BT_GATT_POOL_CHRC/CCC/SVC, none of which take in an initializer argument, to make the API more consistent.

I haven't seet BT_GATT_POOL_DESC used in the way you describe. but it would be possible to do the same with the underlying call `bt_gatt_pool_desc_alloc`
```
bt_gatt_attr = BT_GATT_DESCRIPTOR(...);
bt_gatt_pool_desc_alloc(gp, attr);
```

Reply:
Good point.